LiteCore/Query/QueryParser.cc

```diff
@@ -320,9 +320,12 @@ namespace litecore {
 
 
     void QueryParser::writeStringLiteralAsProperty(slice str) {
-        require(str.size > 0 && str[0] == '.',
-                "Invalid property name '%.*s'; must start with '.'", SPLAT(str));
-        str.moveStart(1);
+        require(str.size > 0 && (str[0] == '.' || str[0] == '*'),
```

Comment:
Yeah I knew that when I wrote it, I was just wondering how often that would happen.  Well, actually I think I can change it to check that the size is also 1.